### PR TITLE
Display proper error message while deleting networks

### PR DESCRIPTION
**Pull request by ravening**
### Description
Display exception message if the networks cant be deleted

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [X] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [X] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
This method has usage in 5 different manager implementations:
[ ] KubernetesClusterDestroyWorker
[ ] NetworkServiceImpl
[ ] VpcManagerImpl
[ ] AccountManagerImpl
[ ] DomainManagerImpl

As this changes the semantics of the call it needs thorough validation in all those components.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java [hunk 1]
```diff
@@ -2882,8 +2882,9 @@ public boolean destroyNetwork(final long networkId, final ReservationContext con
 
         for (final UserVmVO vm : userVms) {
             if (!(vm.getState() == VirtualMachine.State.Expunging && vm.getRemoved() != null)) {
-                s_logger.warn("Can't delete the network, not all user vms are expunged. Vm " + vm + " is in " + vm.getState() + " state");
-                return false;
+                final String message = "Can't delete the network, not all user vms are expunged. Vm " + vm + " is in " + vm.getState() + " state";
+                s_logger.warn(message);
+                throw new InvalidParameterValueException(message);
```
Comment (harikrishna-patnala) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:2887]: Throwing exceptions will cause other issues here, as already mentioned by Daan, other components are handling the boolean response either by logging error message or cleaning up some other resources upon error (AccountManagerImpl).

Comment (sureshanaparti) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java:2887]: > Throwing exceptions will cause other issues here, as already mentioned by Daan, other components are handling the boolean response either by logging error message or cleaning up some other resources upon error (AccountManagerImpl).

@ravening any update?

**Pull request closed without merging**